### PR TITLE
[5.1] Add the ability to set a directory prefix/root to a S3 disk

### PR DESCRIPTION
**Pull request by claesjohansson**
Add missing ability to set a directory prefix to a S3 disk.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
You broke the build.

**Comment by GrahamCampbell** [comment]
Also, I don't think `root` is a good name. `prefix` would be better.

**Comment by claes-gm** [comment]
Using the same markup as local disks in config/filesystems.php, not a good idea ?

For example:
'local' => [
         'driver' => 'local',
         'root' => storage_path() . '/app',
      ],

**Comment by GrahamCampbell** [comment]
Maybe it's fine. Taylor can always change it if he prefers `prefix`.

**Comment by GrahamCampbell** [comment]
Can you fix the cs issues please.

**Comment by claes-gm** [comment]
Fixed cs issue, all checks passed.

File: src/Illuminate/Filesystem/FilesystemManager.php [hunk 1]
```diff
@@ -156,7 +156,7 @@ public function createS3Driver(array $config)
         }
 
         return $this->adapt(
-            new Flysystem(new S3Adapter(new S3Client($config), $config['bucket']))
+            new Flysystem(new S3Adapter(new S3Client($config), $config['bucket'], ! empty($config['root']) ? $config['root'] : ''))
```
Comment (GrahamCampbell) [inline src/Illuminate/Filesystem/FilesystemManager.php]: Would be better to ditch the `!` and swap the stuff either side of the `?`.

**Comment by claes-gm** [comment]
Done!

**Pull request merged**